src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/_validators.py

```diff
@@ -51,39 +22,3 @@ def validate_deployment_name(namespace):
             namespace.deployment_name = os.path.splitext(template_filename)[0]
         else:
             namespace.deployment_name = 'deployment1'
-
-def _resolve_api_version(rcf, resource_type, parent=None):
-
-    provider = rcf.providers.get(resource_type.namespace)
-    resource_type_str = '{}/{}'.format(parent.type, resource_type.type) \
-        if parent else resource_type.type
-
-    rt = [t for t in provider.resource_types if t.resource_type == resource_type_str]
-    if not rt:
-        raise IncorrectUsageError('Resource type {} not found.'
-                                  .format(resource_type_str))
-    if len(rt) == 1 and rt[0].api_versions:
-        npv = [v for v in rt[0].api_versions if 'preview' not in v.lower()]
-        return npv[0] if npv else rt[0].api_versions[0]
-    else:
-        raise IncorrectUsageError(
-            'API version is required and could not be resolved for resource {}/{}'
-            .format(resource_type.namespace, resource_type.type))
-
-def resolve_resource_parameters(namespace):
```

Comment:
If we maintain the ability to specify `resource_type` as 'provider/type' then where do we do that sniffing/parsing/checking?